pickhardtpayments/OracleLightningNetwork.py

```diff
@@ -71,3 +78,23 @@ def theoretical_maximum_payable_amount(self, source: str, destination: str, base
 
         mincut, _ = nx.minimum_cut(test_network, source, destination)
         return mincut
+
+    def settle_payment(self, path: List[Channel], payment_amount: int):
+        """
+        receives a dictionary with channels and payment amounts and adjusts the balances of the channels along the path.
+
+        settle_payment should only be called after all send_onions for a payment terminated successfully!
+        # TODO testing
+        """
+        for channel in path:
```

Comment:
what if the path is not connected? Maybe we really should have a `Path` class.

Reply:
i changed it so that path from an Attempt is passed to settle_payment. At initialisation of the Attempt with the path, the connectedness is checked. 
Can still be refactored to a Path Class later on.